src/einsteinpy/tests/test_metric/test_kerrnewman.py

```diff
@@ -0,0 +1,118 @@
+import warnings
+
+import numpy as np
+import pytest
+from astropy import units as u
+from numpy.testing import assert_allclose
+
+from einsteinpy import constant
+from einsteinpy.metric import KerrNewman
+from einsteinpy.utils import kerrnewman_utils, schwarzschild_radius
+
+_c = constant.c.value
+_G = constant.G.value
+_cc = constant.coulombs_const.value
+
+
+def test_calculate_trajectory0():
+    # Based on the revolution of earth around sun
+    # Data from https://en.wikipedia.org/wiki/Earth%27s_orbit
+    # Initialialized with cartesian coordinates
+    # Function returning cartesian coordinates
+    M = 1.989e30 * u.kg
+    q = 0 * u.C / u.kg
+    distance_at_perihelion = 147.10e6 * u.km
+    speed_at_perihelion = 30.29 * u.km / u.s
+    pos_vec = [
+        distance_at_perihelion / np.sqrt(2),
+        distance_at_perihelion / np.sqrt(2),
+        0 * u.km,
+    ]
+    vel_vec = [
+        -1 * speed_at_perihelion / np.sqrt(2),
+        speed_at_perihelion / np.sqrt(2),
+        0 * u.km / u.h,
+    ]
+    end_lambda = ((1 * u.year).to(u.s)).value
+    cl = KerrNewman.from_cartesian(pos_vec, vel_vec, q, 0 * u.min, M, 0.0, 0 * u.C)
+    ans = cl.calculate_trajectory(
+        start_lambda=0.0,
+        end_lambda=end_lambda,
+        return_cartesian=True,
+        OdeMethodKwargs={"stepsize": end_lambda / 1.5e3},
+    )[1]
+    # velocity should be 29.29 km/s at apehelion(where r is max)
+    R = np.sqrt(ans[:, 1] ** 2 + ans[:, 2] ** 2 + ans[:, 3] ** 2)
+    i = np.argmax(R)  # index whre radial distance is max
+    v_apehelion = (
+        (np.sqrt(ans[i, 5] ** 2 + ans[i, 6] ** 2 + ans[i, 7] ** 2) * (u.m / u.s)).to(
+            u.km / u.s
+        )
+    ).value
+    assert_allclose(v_apehelion, 29.29, rtol=0.01)
+
+
+def test_calculate_trajectory1():
+    # the test particle should not move as gravitational & electromagnetic forces are balanced
+    M = 0.5 * 5.972e24 * u.kg
```

Comment:
Can you just make a pytest fixture for this data? 
You can have a look at an example here: https://github.com/einsteinpy/einsteinpy/blob/master/src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py#L12

Reply:
Sure. No problems if I would do it with `pytest.mark.parametrize` ?